### PR TITLE
Create tests, change AppModel Schema

### PR DESCRIPTION
**Pull request by olegtsigulov**

### CODE REVIEW

File: models/UserModel.js [hunk 1]
```diff
@@ -182,11 +182,11 @@ const updateOne = async function ( condition, updateData ) {
 
 const increaseCountPosts = async ( author ) => {
     try{
-        await UserModel.updateOne(
+        let result = await UserModel.updateOne(
```
Comment (maksimoancha) [inline models/UserModel.js]: use const instead let

File: models/wObjectModel.js [hunk 2]
```diff
@@ -13,32 +13,38 @@ const create = async function ( data ) {
...
+        return { result: result.nModified === 1 };
     } catch ( error ) {
         return { error };
     }
 };
 
 const increaseFieldWeight = async function ( data ) { // data include: author, permlink, author_permlink, weight
+    if ( !data || !data.author || !data.author_permlink ) {
+        return( { error: { message: 'data didnt contains author or author permlink' } } );
+    }
```
Comment (maksimoancha) [inline models/wObjectModel.js]: edit check input data with safety methods lodash (ex. _.every(requiredKeys, _.partial(_.has, inputData)))

File: models/wObjectModel.js [hunk 3]
```diff
@@ -47,33 +53,36 @@ const increaseFieldWeight = async function ( data ) { // data include: author, p
...
+        return { result: result.nModified === 1 };
     } catch ( error ) {
         return { error };
     }
 };
 
 const increaseWobjectWeight = async function ( data ) {
+    if ( !data || !data.author_permlink ) {
+        return( { error: { message: 'Data dont contains author permlink' } } );
+    }
```
Comment (maksimoancha) [inline models/wObjectModel.js]: edit check input data with safety methods lodash (ex. _.every(requiredKeys, .partial(.has, inputData)))

File: package.json [hunk 4]
```diff
@@ -26,6 +26,7 @@
     "lodash": "^4.17.5",
     "mongoose": "^5.3.5",
     "morgan": "~1.9.0",
+    "read": "^1.0.7",
```
Comment (maksimoancha) [inline package.json]: delete unused package

File: parsers/followObjectParser.js [hunk 5]
```diff
@@ -7,26 +7,29 @@ const parse = async function ( data ) {
...
             if( !wobject || error ) {
                 console.log( error || `User ${json[ 1 ].user} try to follow non existing wobject: ${json[ 1 ].author_permlink}` );
-                return;
+                return ( error || `User ${json[ 1 ].user} try to follow non existing wobject: ${json[ 1 ].author_permlink}` );
             }
             const { result } = await User.addObjectFollow( json[ 1 ] );
 
             if ( result ) {
                 console.log( `User ${json[ 1 ].user} now following wobject ${json[ 1 ].author_permlink}!\n` );
+                return `User ${json[ 1 ].user} now following wobject ${json[ 1 ].author_permlink}!\\n`;
```
Comment (maksimoancha) [inline parsers/followObjectParser.js]: duplicate lines, put data to variable

File: parsers/followObjectParser.js [hunk 6]
```diff
@@ -7,26 +7,29 @@ const parse = async function ( data ) {
...
             }
         } else { // else if missing - unfollow
             const { result } = await User.removeObjectFollow( json[ 1 ] );
 
             if ( result ) {
                 console.log( `User ${json[ 1 ].user} now unfollow wobject ${json[ 1 ].author_permlink} !\n` );
+                return ( `User ${json[ 1 ].user} now unfollow wobject ${json[ 1 ].author_permlink} !\n` );
+
```
Comment (maksimoancha) [inline parsers/followObjectParser.js]: delete extra line, put duplicate data to variable

File: test/factories/App/AppFactory.js [hunk 7]
```diff
@@ -1,12 +1,22 @@
...
+        daily_chosen_post: {
+            author: author || faker.name.firstName().toLowerCase(),
+            permlink: permlink || getRandomString(),
+            title: title || getRandomString( 20 )
+        },
+        weekly_chosen_post: {
+            author: author || faker.name.firstName().toLowerCase(),
+            permlink: permlink || getRandomString(),
+            title: title || getRandomString( 20 )
+        }
```
Comment (maksimoancha) [inline test/factories/App/AppFactory.js]: edit param variables name to separate author and permlink between  daily and weekly chosen posts

File: test/factories/Wobjects/wobjectFactory.js [hunk 8]
```diff
@@ -0,0 +1,9 @@
+const { User, faker, getRandomString, WObject } = require( '../../testHelper' );
+const UserFactory = require( '../../factories/User/UserFactory' );
+const ObjectFactory = require( '../../factories/Object/ObjectFactory' );
+
+
+// const Create = async( { name, author_permlink: root_permlink, object_type } = {} ) => {
+
+
+// });
```
Comment (maksimoancha) [inline test/factories/Wobjects/wobjectFactory.js]: delete duplicate factory

File: test/models/WobjectModel/getOne-remoteVoteTest.js [hunk 9]
```diff
@@ -0,0 +1,301 @@
...
+    } );
+    describe( 'On increseWobjectWeight', async () => {
+        let result, data, objType, objTypeName;
+        beforeEach( async () => {
+            objTypeName = getRandomString();
+            data = {
+                author_permlink: getRandomString(),
+                weight: 100
+            };
+            await ObjectFactory.Create( { author_permlink: data.author_permlink, object_type: objTypeName } );
```
Comment (maksimoancha) [inline test/models/WobjectModel/getOne-remoteVoteTest.js]: no need to generate objectTypeName here

File: test/models/WobjectModel/getOne-remoteVoteTest.js [hunk 10]
```diff
@@ -0,0 +1,301 @@
...
+        } );
+        it( 'should success increase wobject weight return true', async () => {
+            result = await WobjModel.increaseWobjectWeight( data );
+            expect( result.result ).is.true;
+        } );
+        it( 'should success increase wobject weight by data weight', async () => {
```
Comment (maksimoancha) [inline test/models/WobjectModel/getOne-remoteVoteTest.js]: unreadable test descrioption

File: test/models/WobjectModel/getOne-remoteVoteTest.js [hunk 11]
```diff
@@ -0,0 +1,301 @@
...
+            await WobjModel.increaseWobjectWeight( data );
+            result = await WObject.findOne( { author_permlink: data.author_permlink } );
+            expect( result.weight ).to.deep.eq( data.weight );
+        } );
+        it( 'should seccess increase object type weight be data weight', async () => {
```
Comment (maksimoancha) [inline test/models/WobjectModel/getOne-remoteVoteTest.js]: unreadable test description

File: test/models/WobjectModel/getOne-remoteVoteTest.js [hunk 12]
```diff
@@ -0,0 +1,301 @@
...
+            await WobjModel.increaseWobjectWeight( data );
+            result = await ObjectType.findOne( { name: objTypeName } );
+            expect( result.weight ).to.deep.eq( data.weight );
+        } );
+        it( 'should eq wobject weight and wobject type weight', async () => {
+            await WobjModel.increaseWobjectWeight( data );
+            result = await WObject.findOne( { author_permlink: data.author_permlink } );
```
Comment (maksimoancha) [inline test/models/WobjectModel/getOne-remoteVoteTest.js]: incorrect variable name

File: test/models/WobjectModel/getOne-remoteVoteTest.js [hunk 13]
```diff
@@ -0,0 +1,301 @@
...
+                    ]
+            };
+            data = {
+                author: field.author,
+                permlink: field.permlink,
+                author_permlink: getRandomString(),
+                weight: 1111,
+                voter: voter
+            };
+            await ObjectFactory.Create( { appends: [ field ], author_permlink: data.author_permlink } );
```
Comment (maksimoancha) [inline test/models/WobjectModel/getOne-remoteVoteTest.js]: use appendObjectFactory instead

File: test/models/WobjectModel/getOne-remoteVoteTest.js [hunk 14]
```diff
@@ -0,0 +1,301 @@
...
+        } );
+        it( 'should success removeVote return true', async () => {
+            result = await WobjModel.removeVote( data );
+            expect( result.result ).is.true;
+        } );
+        it( 'should success remove Vote return false', async () => {
```
Comment (maksimoancha) [inline test/models/WobjectModel/getOne-remoteVoteTest.js]: edit test description

File: test/models/WobjectModel/getOne-remoteVoteTest.js [hunk 15]
```diff
@@ -0,0 +1,301 @@
...
+            result = await WobjModel.removeVote( { author: getRandomString(),
+                permlink: field.permlink,
+                author_permlink: getRandomString(),
+                weight: 1111,
+                voter: voter } );
+            expect( result.result ).is.false;
+        } );
+        it( 'should success remove Vote ', async () => {
```
Comment (maksimoancha) [inline test/models/WobjectModel/getOne-remoteVoteTest.js]: edit test descr.

File: test/models/WobjectModel/m0ck.js [hunk 16]
```diff
@@ -0,0 +1,6 @@
+
+const throwError = async () => {
+    throw new MongooseError.ValidatorError( 'Some problems' );
```
Comment (maksimoancha) [inline test/models/WobjectModel/m0ck.js]: delete unused code

File: test/parsers/followObjectParser/test.js [hunk 17]
```diff
@@ -0,0 +1,52 @@
+const { Mongoose, expect, getRandomString, faker } = require( '../../testHelper' );
```
Comment (maksimoancha) [inline test/parsers/followObjectParser/test.js]: edit tests with stubing model functions

File: test/validators/appendObjectValidator/test.js [hunk 18]
```diff
@@ -21,6 +21,8 @@ describe( 'Append Object Validator', async () => {
 
     } );
 
+    it
```
Comment (maksimoancha) [inline test/validators/appendObjectValidator/test.js]: hello

File: test/models/ObjectTypeModel/test.js [hunk 19]
```diff
@@ -0,0 +1,93 @@
...
+
+
+describe( 'ObjectTypeModel', async () => {
+    describe( 'On getOne', async () => {
+        let objectType2, name;
+        before( async () => {
+            name = getRandomString();
+            objectType2 = await ObjectTypeFactory.Create( {
+                name: name
+            } );
```
[the remark this answers is not in the document]
Comment (olegtsigulov) [reply test/models/ObjectTypeModel/test.js]: I need to generate random name there. I will change it to faker.name

File: test/models/ObjectTypeModel/test.js [hunk 20]
```diff
@@ -0,0 +1,93 @@
...
+
+            expect( res.objectType ).to.deep.eq( objectType2._doc );
+        } );
+    } );
+    describe( 'On create', async () => {
+        let object, result;
+        beforeEach( async () => {
+            object = await ObjectTypeModel.create( { name: getRandomString(), author: getRandomString(),
+                author_permlink: getRandomString()
+            } );
```
[the remark this answers is not in the document]
Comment (olegtsigulov) [reply test/models/ObjectTypeModel/test.js]: it is not a factory, it is real method

File: test/models/PostModel/test.js [hunk 21]
```diff
@@ -0,0 +1,114 @@
+const { expect, PostModel, Post, Mongoose, getRandomString } = require( '../../testHelper' );
+const { PostFactory } = require( '../../factories' );
+
+describe( 'PostModel', async () => {
+    describe( 'On getPostsRefs', async () => {
+        let postModel, postModel2;
+        beforeEach( async () => {
+            await Mongoose.connection.dropDatabase( );
+            postModel = await PostFactory.Create();
+            postModel2 = await PostFactory.Create();
```
[the remark this answers is not in the document]
Comment (olegtsigulov) [reply test/models/PostModel/test.js]: getPostsRefs method implies to get many post refs

File: test/models/PostModel/test.js [hunk 22]
```diff
@@ -0,0 +1,114 @@
...
+        } );
+        it( 'should success findOne post exist', async () => {
+            const foundedPost = await PostModel.findOne( { author: post.author, permlink: post.permlink } );
+
+            expect( foundedPost.post ).to.exist;
+        } );
+        it( 'should success findOne post is null', async () => {
+            const foundedPost = await PostModel.findOne( { author: 'some author', permlink: 'some permlink' } );
+
+            expect( foundedPost.post ).is.null;
```
[the remark this answers is not in the document]
Comment (olegtsigulov) [reply test/models/PostModel/test.js]: not work with to.not.exist, only not.exist

File: test/models/PostModel/test.js [hunk 23]
```diff
@@ -0,0 +1,114 @@
...
+            expect( result_update ).is.exist;
+        } );
+        it( 'should equal  ', async () => {
+            expect( { net_votes: upd_post._doc.net_votes, total_vote_weight: upd_post.total_vote_weight
+            } ).to.not.deep.eq( { net_votes: post.net_votes, total_vote_weight: post.total_vote_weight } );
+        } );
+        it( 'should return error', async () => {
+            let postModel2 = await PostModel.update();
+
+            expect( 'Cannot read property \'author\' of undefined' ).to.deep.eq( postModel2.error.message );
```
[the remark this answers is not in the document]
Comment (olegtsigulov) [reply test/models/PostModel/test.js]: cant do this, dont work

File: test/models/UserModel/increaseWobjectTest.js [hunk 24]
```diff
@@ -0,0 +1,169 @@
...
+
+describe( 'User Model', async () => {
+    describe( 'On increaseWobjectsWeight', async () => {
+        let data, data2, weightIncrease, name;
+        before( async() => {
+            await Mongoose.connection.dropDatabase();
+            name = getRandomString();
+            await UserFactory.Create( { name: name } );
+            data = {
+                name: name,
```
[the remark this answers is not in the document]
Comment (olegtsigulov) [reply test/models/UserModel/increaseWobjectTest.js]: dont understand

**Pull request merged**